modules/core/src/main/scala/lucuma/odb/api/repo/ObservationRepo.scala

```diff
@@ -154,5 +159,31 @@ object ObservationRepo {
           Tables.constraintSetObservation,
           ConstraintSetModel.ConstraintSetEvent.updated
         )
+
+      override def unsetConstraintSet(oid: Observation.Id)(implicit F: MonadError[F, Throwable]): F[ObservationModel] = {
+        val focusObsLens = focusOn(oid)
+
+        val obsLens: Lens[Tables, Option[ObservationModel]] =
+          Lens[Tables, Option[ObservationModel]](focusObsLens.get)(oo => focusObsLens.set(oo))
+
+
+        val doUpdate: F[(ObservationModel, Option[ConstraintSetModel])] = 
+          tablesRef.modify { oldTables => 
+            val obs = obsLens.get(oldTables).toValidNec(InputError.missingReference("id", oid.show))
+            val cs  = 
+              obs.fold(
+                _ => None,
+                o => oldTables.constraintSetObservation.selectLeft(o.id).flatMap(csId => Tables.constraintSet(csId).get(oldTables))
+              )
+            val tables = cs.fold(oldTables)(_ => Tables.constraintSetObservation.modify(_.removeRight(oid))(oldTables))
+            (tables, obs.map(o => (o, cs)))
```

Comment:
I suppose you could `obs.tupleRight(cs)` there if you want.

Reply:
Nice